metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key
+func (m Metafields) GetByKey(key string) (interface{}, error) {
+	var err error
+	for _, metafield := range m {
+		if metafield.Key == key {
+			switch metafield.Type {
+			case NumberIntegerMetaFieldType:
+				var value int64
+				if metafield.Value != "" {
+					value, err = strconv.ParseInt(metafield.Value, 0, 64)
+					if err != nil {
+						return 0, err
+					}
+				}
+
+				return value, nil
+			case BooleanMetaFieldType:
+				var value bool
+				value, err = strconv.ParseBool(metafield.Value)
+				if err != nil {
+					return false, err
+				}
+
+				return value, nil
```

Comment:
Could you not just return this directly?

```suggestion
				return strconv.ParseBool(metafield.Value)
```